src/deluge/model/mod_controllable/mod_controllable_audio.cpp

```diff
@@ -2327,77 +2326,77 @@ void ModControllableAudio::switchDelaySyncType() {
 		delay.syncType = SYNC_TYPE_TRIPLET;
 		break;
 	}
+	display->displayPopup(getDelaySyncTypeDisplayName());
+}
 
-	char const* displayText;
+char const* ModControllableAudio::getDelaySyncTypeDisplayName() {
 	switch (delay.syncType) {
 	case SYNC_TYPE_TRIPLET:
-		displayText = "Triplet";
-		break;
+		return "Triplet";
```

Comment:
not safe - these will be on the stack and not valid after the return. Have them take a char* as input and write the result into it, so the caller manages the memory